File: servicio/Profile/Infrastructure/Dtos/UserDto.cs
using servicio.Profile.Domain;

namespace servicio.Profile.Infrastructure.Dtos;

public class UserDto
{
    public string FirstName {get; set;} = string.Empty;
    public string LastName {get; set;} = string.Empty;
    public string UserName {get; set;} = string.Empty;
    public string Password {get; set;} = string.Empty;
    public string Email {get; set;} = string.Empty;
    public string PhoneNumber {get; set;} = string.Empty;


    public static UserDto FromUser(User user){
        return new UserDto(){
            FirstName = user.FirstName,
            LastName = user.LastName,
            UserName = user.UserName.Value,
            Password = user.Password,
            Email = user.Email.Value,
            PhoneNumber = user.PhoneNumber.Value
        };
    }

    public User ToUser(){
        return new User(
            FirstName, 
            LastName,
            Password, 
            new UserName(UserName), 
            new UserEmail(Email), 
            new UserPhoneNumber(PhoneNumber));
    }
}

File: client/src/Profile/Entities/UserDto.ts
import User from "./User";

export default class UserDto {
    public FirstName:string;
    public LastName:string;
    public UserName:string;
    public Password:string;
    public Email: string;
    public PhoneNumber: string;
    constructor(firstName:string, lastName:string, username:string, password:string, email: string, phoneNumber: string) {
        this.FirstName = firstName;
        this.UserName = username;
        this.Password = password;
        this.LastName = lastName;
        this.Email = email;
        this.PhoneNumber =  phoneNumber;
    }

    toUser(): User {
        return new User(this.FirstName, this.LastName, this.UserName, this.Password, this.Email, this.PhoneNumber)
    }

    public static fromUser(user: User): UserDto{
        return new UserDto(user.FirstName, user.LastName, user.UserName, user.Password, user.Email, user.PhoneNumber);
    }
}